### ISSUE TITLE
Puppet support

### ISSUE THREAD

**Issue by cleonte** [jtanx/ctagsx#14]
Great module! i'm using it for puppet, it works for most of the stuff but somehow dose not work for a class like this:

settings::selinux , any pointers?

Cheers,
Cristi

**Comment by jtanx** [comment]
ctagsx just parses tag files to know where definitions are. What are you using to generate it? Can you see a corresponding entry in the tags file? There's [universal ctags](https://github.com/universal-ctags/ctags), which looks like it supports puppet manifests.

**Comment by cleonte** [comment]
generating with this ctags --tag-relative --extra=f -R .

the tags files looks like this

exec['setenforce']	modules/tts_settings/manifests/selinux.pp	/^  exec { 'setenforce':$/;"	r
file['/etc/selinux/config']	modules/tts_settings/manifests/selinux.pp	/^  file { '\/etc\/selinux\/config':$/;"	r
selinux.pp	modules/tts_settings/manifests/selinux.pp	1;"	F
tts_settings::selinux	modules/tts_settings/manifests/selinux.pp	/^class tts_settings::selinux {$/;"	c

so there is an entry for the ::selinux class but somehow pressing go to def dose not work

**Comment by jtanx** [comment]
I see the tag is for `tts_settings::selinux`. What is your text selection to jump to the definition? If you're only searching on `selinux`, then it won't work.

**Comment by cleonte** [comment]
i selected whole tts_settings::selinux and it only goes to tts_settings, not class tts_settings::selinux

**Comment by jtanx** [comment]
I'm sorry, I don't know anything about puppet, so without more context, I don't know about your specific circumstance.

At a guess (based off that tag you provided), if you have:

```puppet
class tts_settings::selinux {
```

And it's navigating the caret to 

```puppet
class tts_settings::selinux {
      ^
```

That's kind of expected behaviour. ctagsx knows nothing about classes; it's just a general tag parser. The tag is `tts_settings::selinux`, so it navigates to the start of that tag.

As mentioned before, you can try out other tags generators like universal ctags, which might generate better tags than the vanilla ctags implementation.

**Comment by cleonte** [comment]
think you can close it :) not working with univesal ctags either and not ctagsx extension problem :)

**Comment by cleonte** [comment]
in the end it seems it works! if you select the whole world and prest ctrl+t ! :) thank you 👍

**Comment by dody** [comment]
Hi, thanks for the great work 👍 Your extension is fastest with my 50MB ctags file it takes miliseconds compared to other extensions that takes 5 to 10 seconds.

I have similar issue with TCL language.

Scenario:
- I have proc: `my_namespace::my_proc`
- When I select it and do F12 or 'Go to definition' i will get error: No definition found for 'my_namespace'
- It works when I use ctrl + t

Whole source code:
```tcl
namespace eval my_namespace {
}

proc my_proc {args} {
    # just to confuse ctags and catch bugs
}

proc my_namespace::my_proc {args} {
}

proc test {args} {
    # test the next line
    my_namespace::my_proc
}
```

Also definition peak does not work.

I have attached the code as well as tags file.
[test.zip](https://github.com/jtanx/ctagsx/files/3023264/test.zip)

Please let me know if you have any questions.

**Comment by jtanx** [comment]
This was because via the definition provider, I only get the document and a single position, so I was reliant on `getWordRangeAtPosition`, which just returns `my_namespace`, hence leading to not finding anything. I've fixed this by checking if there's an editor, and preferencing the selection over the position provided. It should be fixed as part of 1.1.1.

**Comment by dody** [comment]
Thanks for such quick response. Working great.
